etcd/src/test/java/org/jclouds/etcd/features/KeysApiMockTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.etcd.features;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+import org.jclouds.etcd.EtcdApi;
+import org.jclouds.etcd.EtcdApiMetadata;
+import org.jclouds.etcd.domain.keys.Key;
+import org.jclouds.etcd.internal.BaseEtcdMockTest;
+import org.testng.annotations.Test;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+/**
+ * Mock tests for the {@link org.jclouds.etcd.features.KeysApi} class.
+ */
+@Test(groups = "unit", testName = "KeysApiMockTest")
+public class KeysApiMockTest extends BaseEtcdMockTest {
+
+   public void testCreateKey() throws Exception {
+      MockWebServer server = mockEtcdJavaWebServer();
+
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/keys-create.json")).setResponseCode(201));
+      EtcdApi etcdApi = api(server.getUrl("/"));
+      KeysApi api = etcdApi.keysApi();
+      try {
+         Key createdKey = api.createKey("hello", "world");
+         assertNotNull(createdKey);
+         assertTrue(createdKey.node().key().equals("/hello"));
+         assertTrue(createdKey.node().value().equals("world"));
+         assertSent(server, "PUT", "/" + EtcdApiMetadata.API_VERSION + "/keys/hello");
```

Comment:
There are no form parameters being sent it's all on the request path:

https://github.com/coreos/etcd/blob/master/Documentation/api.md#setting-the-value-of-a-key